ipv8/src/main/java/nl/tudelft/ipv8/messaging/eva/EVAProtocol.kt

```diff
@@ -0,0 +1,804 @@
+package nl.tudelft.ipv8.messaging.eva
+
+import kotlinx.coroutines.*
+import mu.KotlinLogging
+import nl.tudelft.ipv8.Community
+import nl.tudelft.ipv8.Peer
+import nl.tudelft.ipv8.keyvault.Key
+import nl.tudelft.ipv8.util.toHex
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.util.*
+
+private val logger = KotlinLogging.logger {}
+
+open class EVAProtocol(
+    private var community: Community,
+    scope: CoroutineScope,
+    var windowSizeInBlocks: Int = WINDOW_SIZE_IN_BLOCKS,
+    var retransmitIntervalInSec: Int = RETRANSMIT_INTERVAL_IN_SEC,
+    var retransmitAttemptCount: Int = RETRANSMIT_ATTEMPT_COUNT,
+    var scheduledSendIntervalInSec: Int = SCHEDULED_SEND_INTERVAL_IN_SEC,
+    var timeoutIntervalInSec: Int = TIMEOUT_INTERVAL_IN_SEC,
+    var binarySizeLimit: Int = BINARY_SIZE_LIMIT,
+    var terminateByTimeoutEnabled: Boolean = true,
+    var blockSize: Int = BLOCK_SIZE,
+    var retransmitEnabled: Boolean = true,
+    var loggingEnabled: Boolean = true
```

Comment:
Should probable be off per default.  Also why this additional parameter for logging? We're already using a logger.

Reply:
The logging should indeed be disabled by default. I've included it since the protocol generates quite a lot of log entries, which can easily push out log entries of other parts of the app. In some situations it may come in handy to debug with logs enabled for future improvements. I'll disable it by default after the meeting.

Reply:
fixed in branch